.github/workflows/tests-template.yml

```diff
@@ -42,13 +42,7 @@ jobs:
               CPU=4 make test
               ;;
             linux-unit-test-4-cpu-race)
-              # XXX: By default, the Github Action runner will terminate the process
-              # if it has high resource usage. Try to use GOGC to limit memory and
-              # cpu usage here to prevent unexpected terminating. It can be replaced
-              # with GOMEMLIMIT=2048MiB if the go-version is updated to >=1.19.x.
-              #
-              # REF: https://github.com/actions/runner-images/issues/6680#issuecomment-1335778010
-              GOGC=30 CPU=4 ENABLE_RACE=true make test
+              GOMEMLIMIT=2048MiB CPU=4 ENABLE_RACE=true make test
```

Comment:
Based on the `Suggested uses` in https://tip.golang.org/doc/gc-guide, 
- `Don't use the memory limit when deploying to an execution environment you don't control`.
- `Don't set a memory limit to avoid out-of-memory conditions when a program is already close to its environment's memory limits.`

Also the default value for GOGC is 100, which should be good enough.

What we can do is to try to use larger runner.

So can we just remove the configuration? I mean we don't use GOGC, nor GOMEMLIMIT?

Reply:
The original issue seems to imply jobs are being killed due to excessive resource usage: https://github.com/actions/runner-images/issues/6680 and based on discussion / linked issues it appears to be ongoing.

However I don't actually know if we are still impacted by this, based on the comments we were historically.

Agree let's try large runner for longer jobs and see if we can just remove it altogether.  We can always reinstate the limit if necessary.

Reply:
These changes would change the approach for this PR. I pushed them to #612.

Reply:
The GOGC was used to prevent from unexpected kill. I think the issue is still here if we don't use larger size runner.
Yeah, I prefer to abort this one and use #612.

Reply:
Would using the large runner just for the race test case make sense? And leave the others with the regular runner? I can change the implementation if that sounds good.

Reply:
SGTM - 4cpu race (amd64 +arm64) and windows tests are the ones that need extra resources.